src/EnergyPlus/Material.hh

```diff
@@ -343,18 +384,240 @@ namespace Material {
+        Real64 ReflectSolBeamFront = 0.0;          // Beam solar reflected as diffuse radiation when sun is in front of screen
+        Real64 ReflectVisBeamFront = 0.0;          // Visible solar reflected as diffuse radiation when sun is in front of screen
+        Real64 ReflectSolBeamBack = 0.0;           // Beam solar reflected as diffuse radiation when sun is in back of screen
+        Real64 ReflectVisBeamBack = 0.0;           // Visible solar reflected as diffuse radiation when sun is in back of screen
+        Real64 AbsorpSolarBeamFront = 0.0;         // Front surface solar beam absorptance
+        Real64 AbsorpSolarBeamBack = 0.0;          // Back surface solar beam absorptance
+        Real64 DifDifTrans = 0.0;                  // Back surface diffuse solar transmitted
+        Real64 DifDifTransVis = 0.0;               // Back surface diffuse visible solar transmitted
+        Real64 DifScreenAbsorp = 0.0;              // Absorption of diffuse radiation
+        Real64 DifReflect = 0.0;                   // Back reflection of solar diffuse radiation
+        Real64 DifReflectVis = 0.0;                // Back reflection of visible diffuse radiation
+        Real64 ReflectScreen = 0.0;                // Screen assembly solar reflectance (user input adjusted for holes in screen)
+        Real64 ReflectScreenVis = 0.0;             // Screen assembly visible reflectance (user input adjusted for holes in screen)
+        Real64 ReflectCylinder = 0.0;              // Screen material solar reflectance (user input, does not account for holes in screen)
+        Real64 ReflectCylinderVis = 0.0;           // Screen material visible reflectance (user input, does not account for holes in screen)
+        Real64 ScreenDiameterToSpacingRatio = 0.0; // ratio of screen material diameter to screen material spacing
+        ScreenBeamReflectanceModel screenBeamReflectanceModel =
+            ScreenBeamReflectanceModel::Invalid; // user specified method of accounting for scattered solar beam
+    };
+
+    struct ScreenTransData
+    {
+        // Members
+        Array2D<Real64> Trans;
+        Array2D<Real64> Scatt;
     };
 
     void GetMaterialData(EnergyPlusData &state, bool &errorsFound); // set to true if errors found in input
     void GetVariableAbsorptanceInput(EnergyPlusData &state, bool &errorsFound);
+    std::string DisplayMaterialRoughness(SurfaceRoughness Roughness); // Roughness String
 
 } // namespace Material
 
 struct MaterialData : BaseGlobalStruct
 {
-    EPVector<Material::MaterialProperties *> Material;
-    int TotMaterials = 0; // Total number of unique materials (layers) in this simulation
+    EPVector<Material::MaterialBase *> Material;
+    int TotMaterials = 0;     // Total number of unique materials (layers) in this simulation
+    int TotComplexShades = 0; // Total number of shading materials for complex fenestrations
 
+    Array1D<Material::WindowBlindProperties> Blind;
```

Comment:
Should this also be an EPVector?

Reply:
Yeah. There's one redimension adding a new blind, but that can be done in EPVector too